File: InfoQ/35.py
<html><head><meta http-equiv="content-type" content="text/html; charset=utf-8" /></head><body><h3>Elasticsearch 1.0.0リリース</h3><p><a target="_blank" href="http://www.infoq.com/news/2014/02/elasticsearch_1.0.0_released"><em>原文(投稿日：2014/02/14)へのリンク</em></a></p>
<div class="article_page_left news_container text_content_container"> 
 <div class="text_info"> 
  <p>オープンソースアナリティクスツールであるElasticsearchのバージョン1.0.0が<a href="http://www.elasticsearch.org/blog/1-0-0-released/">リリース</a>された。Elasticsearchはビッグデータ環境でのリアルタイム分析を実現する分散検索エンジンだ。<a href="http://lucene.apache.org/">Apache Lucene</a>を利用し、ReST APIを公開することで実現している。HTTP経由で直接アクセスできる以外に、Java、JavaScript、Pythonなどの言語向けのライブラリからもアクセスできる。また、<a href="http://hadoop.apache.org/">Apache　Hadoop</a>とも統合できる。Elasticsearchはすでに大規模データを扱っている企業で採用されている。GitHub、Foursquare、SoundCloudといった企業だ。</p> 
  <p>Elasticsearchは拡張性、高可用性、リアルタイム分析の3つに注力している。データはすぐにインデックスされ、クラスタ内で複製され、分析できる状態になる。</p> 
  <ul> 
   <li><strong>拡張性:</strong> Elasticsearchはクラスタ環境で動作するよう設計されている。ノードがスタートするとすぐに、ノットワーク上の他のノードを探して接続する。インデックスはシャード内に作成され、クラスタ上に分散されるので、インデックスの検索は分散処理になり、並列で行われる。性能が必要な場合はノードをさらに追加するだけで自動的にシャードが再構成される。</li> 
   <li><strong>可用性:</strong> データベースシャードは水平的なスケーリングだけでなく、可用性のためにも使われている。各シャードは異なるクラスタに保存されているので、ノードが落ちても、データは失われない。動作不良になったノードはElasticsearchが検知してクラスタから除外する。除外した後、シャードは弾力性と拡張性を考慮して再構成される。<br /> 全クラスタの再起動をサポートするため、Elasticsearchに必要なすべてのメタデータはさまざまなストレージ上に保存されている。データはゲートウェイと呼ばれる仕組みを使って保存される。ゲートウェイは現時点ではローカルのストレージか共有ファイルシステムをサポートする。</li> 
   <li><strong>リアルタイム:</strong> Elasticsearchはスキーマにとらわれない。任意のJSONドキュメントを保存できる。ドキュメントの構造は分析される。タイムスタンプのようなデータ型ですら自動的に検知される。既定では、ドキュメントに含まれるすべてのフィールドがインデックスされ検索可能になる。単純な全文検索、ファセットがインデックスに適用される。ファセットとは分析関数でバケッティング(データの幅、距離、ヒストグラムなど)とメトリクス(合計、平均、統計値など)を提供する。</li> 
  </ul> 
  <p>&nbsp;</p> 
  <h3>Elasticsearch 1.0.0の新しい機能</h3> 
  <p>バージョン1.0.0はさまざまな関数の強化とAPIの変更が行われており、Elasticsearchがより直感的に使えるようになっている。また、インデックスのバックアップ、復元、データの分析、Elasticsearchの弾力性の強化などの機能が搭載されている。</p> 
  <ul> 
   <li><strong>スナップショット/リストア:</strong> 新しいバージョンでは、クラスタ全体のスナップショットを取り、バックアップを作成できる、シンプルなAPIがある。メタデータとインデックスを含む、Elasticsearchのクラスタ全体をスナップショットリポジトリに保存できる。組み込みのフェイルオーバシステムが正常に動作しない場合、クラスタをスナップショットから復元できる。</li> 
   <li><strong>アグリゲーション:</strong> アグリゲーションによって、バージョン1.0.0以前のファセットよりも強力なデータ分析ができる。ファセットは分析関数の結果だけを提供するが(例えば、ある距離に含まれている店舗の数)、アグリゲーションは特定の問い合わせでどのドキュメントが見つかったかを保持し、新しい問い合わせのインプットにそれらのドキュメントを利用できる(例えば、ある距離に含まれている店舗の四半期の平均売り上げ)。</li> 
   <li><strong>サーキットブレーカー:</strong> サーキットブレーカーは検索インデックスに致命的な被害を与える処理エラーや実行エラーを回避するために追加された。Elasticsearch1.0.0に追加された最初のセーフガードはフリーメモリを監視し、検索と分析に必要なメモリ量を見積もる仕組みだった。検索や分析に必要なメモリが利用できるメモリを上回るとき、処理をブロックしてOutOfMemory例外が発生しないようにする。将来のリリースではより多くの種類のサーキットブレーカーが追加される予定。</li> 
  </ul> 
  <p>&nbsp;</p> 
  <p>Elasticsearchはメジャーバージョンを変え、既存のAPIをきれいに整理した。後方互換性は維持されていない。バージョン1.0.0にアップグレードする前に、ユーザは<a href="http://www.elasticsearch.org/guide/en/elasticsearch/reference/1.x/breaking-changes.html">破壊的な変更</a>一覧にあるすべてのデータをバックアップする必要がある。</p> 
  <p>またElasticsearchはデータのキャプチャと分析に使えるツールも提供している。<a href="http://www.elasticsearch.org/overview/logstash">Logstash</a>と<a href="http://www.elasticsearch.org/overview/kibana">Kibana</a>を使うと、Elasticsearchは<a href="http://www.infoq.com/news/2014/02/">ELK-stack</a>として使え、ログファイルを解析し、分析しさまざまな方法で視覚化できる。<br /> また、<a href="http://www.elasticsearch.com/">商用ブランチ</a>経由でサポートを購入することもできる。</p> 
 </div> 
</div><br><br><br><br><br><br></body></html>

File: InfoQ/35.java
<html><head><meta http-equiv="content-type" content="text/html; charset=utf-8" /></head><body><h3>Apcera Continuum</h3><p>Cloud startup Apcera is making its Continuum product more visible with the launch of a new <a href="http://apcera.com/">website</a> on 14 Nov.</p>
<p>Continuum will be taking on Pivotal’s Cloud Foundry open source PaaS, which Apcera founder Derek Collison helped to create. Whilst Pivotal has enabled integration with existing enterprise infrastructure from VMware with the recent release of its commercial <a href="http://www.infoq.com/news/2013/11/pivotal-cf">Pivotal CF</a>, Apcera are focussing on a more holistic policy based approach. In an introductory blog post ‘<a href="http://apcera.com/blog/why-were-here/">why we’re here</a>’ Collison writes :</p>
<blockquote>
 For enterprise IT to out-innovate its competitors, it needs to put policy front and center, successfully blend delivery models (IaaS, PaaS, SaaS) in one platform, and enable semantic awareness of communications. 
 <div>
  &nbsp;
 </div> 
 <div>
  Today, PaaS is still a first-generation cloud-computing model, framed around making developers’ lives easier. What PaaS doesn’t do is address critical enterprise needs such as governance, policy, compliance, authentication, identity, security, auditing, etc. The more we see people using PaaS, the more we understand its limitations. There has never been a single, fully enterprise-grade platform that makes all delivery models work together in compliance while accelerating both Dev and Ops. Up until now, it wasn’t clear how that could happen.
 </div> 
</blockquote>
<p>Apcera have chosen not to go down an open source route to market, though they have open sourced some components such as <a href="https://groups.google.com/forum/#!topic/natsio/amSefF3TbDI">gnatsd</a> ‘a high performance NATS server’, which is used within Continuum for messaging. Collison has promised to open source more of the product over time, but explains why Continuum isn’t fully open source:</p>
<blockquote>
 Ecosystem engagement is a delicate balance. In my previous efforts from within Google to VMware, there were different tools to try to achieve this goal. A common one that has become table stakes is to open source the tool, software or platform. This does drive growth, but I will argue it does not, in itself, solve the ecosystem engagement and nurturing that is required long term. 
 <div>
  &nbsp;
 </div> For instance, if a system is open source, but is not purposely built to be programmable, extensible, and composeable from the inside out, different members of the ecosystem will drive the platform in different ways, leading to an implicit forking problem, where your version has diverged quickly from the lead member, and presents ever-increasing costs to merging back to the main branch. This can be seen in many popular IaaS and PaaS solutions on the market today that are driving ecosystem engagement primarily through open source. 
</blockquote>
<p>Some of the capabilities that Apcera are demoing for Continuum look a lot like <a href="http://www.infoq.com/news/2013/10/dotcloud-renamed-docker">Docker’s</a> approach to lightweight virtualization. Collison explains what’s actually going on under the hood:</p>
<blockquote>
 Within Continuum, everything to us is just a job, a workload that our platform runs, manages, monitors, and allows users to update, report on and govern. There is no difference in our mind and within our platform between an OS, a legacy app, or a web or mobile application using a modern framework. 
 <div>
  &nbsp;
 </div> These things all need isolation in order to achieve many of the goals mentioned above. In that regard, we at Apcera always speak in terms of Isolation Contexts. Isolation Contexts are always isolated, insulated, and autonomous. We believe they are many ways to achieve this, from full-on virtualization with a hypervisor, to lighter-weight primitives like Linux containers, or Solaris Zones, and even micro-task virtualization that’s driven directly by hardware CPUs as popularized by companies like Bromium. 
 <div>
  &nbsp;
 </div> We implemented our first version using the underlying primitives to Linux containers, and there will be many more. Heterogeneous and hybrid are the norm, and one size does not fit all. Lost in our definition could be autonomous, but this one is very important to us. What we mean by autonomous is that every job within the system can only communicate with endpoints that are allowed and defined by policy. This policy reaches from physical access to the networking layer, to addressing, discovery and semantic awareness of communication protocols as mentioned above. Again, this is an extremely powerful pattern. Some have called this Application SDN. We believe we are the first to have really taken this problem head on. And of course, as a modern platform, this is all transparent to the user, and as things change, move, and restart, Continuum handles all of the undifferentiated heavy lifting for you, while supplying secure audit logs for every state transition within the system. With policy built into the core of the architecture, customers gain visibility, insight and compliance from day one.
</blockquote>
<p>Apcera have been an early and vocal adopter of the <a href="http://www.infoq.com/presentations/Go-Google">Go</a> programming language, which is used extensively throughout the platform. Meanwhile Cloud Foundry have been migrating components such as their router, logging system, health manager and <a href="http://blog.cloudfoundry.com/2013/11/09/announcing-cloud-foundry-cf-v6/">CLI</a> over to Go to take advantage of its suitability for highly concurrent applications.</p><br><br><br><br><br><br></body></html>